src/coreclr/jit/vartype.h

```diff
@@ -139,35 +141,29 @@ inline var_types varTypeUnsignedToSigned(var_types vt)
                 unreached();
         }
     }
-    else
-    {
-        return vt;
-    }
+
+    return type;
 }
 
-// If "vt" is a signed integral type, returns the corresponding unsigned integral type, otherwise
-// return "vt".
-inline var_types varTypeSignedToUnsigned(var_types vt)
+// If "vt" represents a signed integral type, returns the corresponding unsigned integral type,
+// otherwise returns the original type.
+template <class T>
+inline var_types varTypeToUnsigned(T vt)
```

Comment:
If I havee read the description correctly this function is unused and should be deleted?

Reply:
Hmm, not sure what do you mean? This is the replacement for `genUnsignedType` and is used in quite a few places. If you meant that `varTypeSignedToUnsigned` "used to be unused" than that's correct and I did delete it (or I guess it is better to say that I unified the two as they were identical anyways).